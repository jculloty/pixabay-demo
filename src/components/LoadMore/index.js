import React from 'react';
import PropTypes from 'prop-types';

import './load-more.css';

function LoadMore(props) {
  return (
    <div className="load-more-button">
      <button onClick={props.onClick}>
        Load more images...
      </button>
    </div>
  );
}

LoadMore.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default LoadMore;
