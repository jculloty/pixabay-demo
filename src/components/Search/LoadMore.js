import React from 'react';

import withApi from '../API';

function LoadMore(props) {
  return (
    <div>
      <button onClick={props.loadMore} className="load-more-button">
        Load more images...
      </button>
    </div>
  );
}

export default withApi(LoadMore, ['loadMore']);
