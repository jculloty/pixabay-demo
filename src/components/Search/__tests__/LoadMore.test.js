import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'test-utils/setup-enzyme';
import LoadMore from '../LoadMore';

const buildComponent = function(props = {}) {
  props = {
    ...{
      value: 'testing',
      handleChange: () => {},
      handleEnter: () => {},
    },
    ...props,
  };

  return (<LoadMore {...props} />);
};

describe('Search: LoadMore', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(buildComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Write more tests', () => {
    expect(true).toBe(false);
  });
});
