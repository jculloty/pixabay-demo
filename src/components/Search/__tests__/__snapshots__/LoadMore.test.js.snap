// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search: LoadMore renders correctly 1`] = `
<div>
  <button
    className="load-more-button"
    onClick={[Function]}
  >
    Load more images...
  </button>
</div>
`;
